web/src/date-utils.ts

```diff
@@ -52,33 +31,36 @@ const isDateValid = (date: Date | number) => {
 
 export const dateToFormat = (date: Date | number, format: DateFormat): string => {
   if (isDateValid(date)) {
+    const dateObject = new Date(date);
+    const hours = padLeadingZero(dateObject.getHours());
+    const minutes = padLeadingZero(dateObject.getMinutes());
+    const seconds = padLeadingZero(dateObject.getSeconds());
+
     switch (format) {
       case DateFormat.TimeShort:
-        return timeShortFormatter.format(date);
-        break;
+        return `${hours}:${minutes}`;
       case DateFormat.TimeMed:
-        return timeMedFormatter.format(date);
-        break;
+        return `${hours}:${minutes}:${seconds}`;
       case DateFormat.DateMed: {
-        const parts = dateMedFormatter.formatToParts(date);
-        return `${getPart(parts, 'year')}-${getPart(parts, 'month')}-${getPart(parts, 'day')}`;
+        const month = padLeadingZero(dateObject.getMonth() + 1);
+        const date = padLeadingZero(dateObject.getDate());
```

Comment:
We already have a `date` declared in the outer scope, so we should call this something different.

Reply:
We can add an `'no-shadow': ['error']` eslint rule to catch these.